src/client_metrics.rs

```diff
@@ -210,6 +283,36 @@ impl ImpactMetricEnv {
     }
 }
 
+impl Merge for ImpactMetricEnv {
+    fn merge(self, other: ImpactMetricEnv) -> ImpactMetricEnv {
```

Comment:
pub trait Merge {
       fn merge(&mut self, other: Self);
   } would be more efficient but mutable so more difficult to reason about